src/routes/category.routes.ts

```diff
@@ -1,19 +1,22 @@
 import express, { Request, Response } from 'express';
+import multer from 'multer';
 import CategoryController from '../controllers/products/category.controller';
+import { validateImageUpload } from '../middlewares/Products/ProductsCategoryMiddleware';
 
 const router = express.Router();
+const upload = multer({ storage: multer.memoryStorage() });
 const categoryController = CategoryController.getInstance();
 
-// CREATE a new category
-router.post('/', async (req: Request, res: Response) => {
+// CREATE a new category with image upload
+router.post('/upload', upload.single('image'), async (req: Request, res: Response) => {
```

Comment:
bro don't override the category?

create this a new end point and also add the swaager for this